### DISCUSSION TITLE
How to reload a relayPagination list?

### DISCUSSION THREAD

**Discussion by schpet** [urql-graphql/urql#1455] [Q&A] [answered]
given I have am grabbing a list of nodes with something like

```graphql
  query SearchResults($after: String, $first: Int, $query: String) {
    search(first: $first, after: $after, query: $query) {
      nodes {
        ...SomeFragment
      }
      pageInfo {
        endCursor
        hasNextPage
        hasPreviousPage
      }
    }
  }
```

and `search` uses `relayPagination` – i might come to my screen and see the first page of results, then infinite scroll a bit and get the second page of results via providing the `after` variable. later when i come back to this screen, i would like to see a full list of new results, but it seems like the cache stays in tact giving me the 2nd page from the old list. how do i get a fresh version of this list of nodes, without the old cached ones? ideally with something like `cache-and-network` so i can show the old results while it loads.

**Comment by kitten** [comment, marked as answer]
So, while `relayPagination` is a great pattern to quickly try out infinite scrolling I'd actually recommend a fragment based approach, ie a component that keeps some page state and a callback where children components render fragments of content with query hooks. This way you get more control over this behaviour.

If you wanted to use `relayPagnation` you'd have to start over with the page variables but even then you'd unfortunately still have some stale data and in the worst case (due to cursors) corrupted pages

**Reply by schpet** [reply]
thanks for this recommendation! in my case i'm on react-native using a flatlist, but i think i get the the approach: store the list in the component state instead of relying on the cache to have it.

**Reply by schpet** [reply]
hmmm realizing this isn't really that easy for a flatlist- it seems to want a list of nodes. trying to think of ways around this, e.g. a flatlist that renders an entire page as an item (possibly rendering a flatlist from within a flatlist?)

since i can't run an arbitrary number of hooks, is there anyway that urql can give me results for multiple useQuery calls so that i could assemble a big list of nodes?

**Reply by kitten** [reply]
I think ultimately it may be easier for us to look into ways in which a `relayPagination` utility can be used, while still being able to invalidate pages. I'll have a look, but I'm aware that this is a little tricky with React Native since it basically requires a flat list of items for `FlatList`

**Reply by schpet** [reply]
really appreciate your attention, been very stumped on this! 🙇🏻‍♂️

**Reply by schpet** [reply]
@kitten i'm finally revisiting this issue and curious if things have changed and if there's any approach you'd currently recommend for getting the data for a flatlist. i might explore messing with a local resolver or something but thought i'd hit you up first in case you happen to know a better way to handle this now.
